engine/classes/ElggIcon.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+/**
+ * Entity icon class
+ */
+class ElggIcon extends ElggFile {
```

Comment:
I think there is a need for an `after` hook in `saveIcon`, the file plugin can add it's metadatas and others can put stuff to river etc, or just convert icons to entities.